lib/portage/package/ebuild/doebuild.py

```diff
@@ -2383,7 +2383,7 @@ def _check_build_log(mysettings, out=None):
     # Configuration:
     #  Automake:                   ${SHELL} /var/tmp/portage/dev-libs/yaz-3.0.47/work/yaz-3.0.47/config/missing --run automake-1.10
     am_maintainer_mode_re = re.compile(
-        r"/missing( --run|'|) (automake|autoconf|autoheader|aclocal)"
+        r"cd.*&&.*/missing( --run|'|) (automake|autoconf|autoheader|aclocal)"
```

Comment:
Let's ensure `cd` isn't part of a longer word.

```suggestion
        r"\bcd\b.*&&.*/missing( --run|'|) (automake|autoconf|autoheader|aclocal)"
```

Reply:
Done, makes sense.